ui/src/routes.tsx

```diff
@@ -31,24 +31,49 @@ import RoleDetails from "./containers/roles.list/details";
 import NewUser from "./containers/users.create";
 import Users from "./containers/users.list";
 import UserDetails from "./containers/users.list/details";
+import * as R from "ramda";
+import UnauthorizedState from "./components/states/Unauthorized";
+import LoadingState from "./components/states/Loading";
 
 export default memo(() => {
-  const { client, user } = useFrontier();
+  const { client, user, isUserLoading } = useFrontier();
+  const [isOrgListLoading, setIsOrgListLoading] = useState(false);
   const [isAdmin, setIsAdmin] = useState(false);
 
+  const isUserEmpty = R.either(R.isEmpty, R.isNil)(user);
+
   useEffect(() => {
-    try {
-      async function getOrganizations() {
-        await client?.adminServiceListAllOrganizations();
-        setIsAdmin(true);
+    async function getOrganizations() {
+      setIsOrgListLoading(true);
+      try {
+        const resp = await client?.adminServiceListAllOrganizations();
+        if (resp?.data?.organizations) {
+          setIsAdmin(true);
+        }
+      } catch (error) {
+        setIsAdmin(false);
+      } finally {
+        setIsOrgListLoading(false);
       }
+    }
+
+    if (!isUserEmpty) {
       getOrganizations();
-    } catch (error) {
-      setIsAdmin(false);
     }
-  }, []);
+  }, [client, isUserEmpty]);
+
+  const isLoading = isOrgListLoading || isUserLoading;
 
-  return user && isAdmin ? (
+  return isLoading ? (
```

Comment:
[Optional] I think we should use `if..else` statements here instead of nested ternary operators. The current implementation is a bit difficult to read.